src/Components/MainMenu/MainMenu.js

```diff
@@ -140,15 +140,24 @@ class MainMenu extends React.Component {
     );
   };
 
+  isDeeplyNested(arr) {
+    let nested = false;
+    arr.map(item => {
```

Comment:
return items.some( item => (item.items ||[]).some((obj => obj.hasOwnProperty('items'))));

you dont want to map you want a some... map will continue the other array even if the first one is nested.